Source/FieldSolver/WarpXExternalEMFields.cpp

```diff
@@ -85,19 +94,37 @@ WarpX::ApplyExternalFieldExcitationOnGrid (
                 amrex::Real x, y, z;
                 WarpXUtilAlgo::getCellCoordinates(i, j, k, mfx_stag,
                                                   problo, dx, x, y, z);
-                Fx(i, j, k) += xfield_parser(x, y, z, t);
+                auto tmp_field_value = xfield_parser(x,y,z,t);
```

Comment:
Maybe we don't ALWAYS want to call the parser.  For most runs when flag_type=-1, this is just wasting CPU time.

if (flag_type >= 0) {
   Fx = flag_type * Fx + parsed_value
}

Reply:
Yes!